src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,289 @@
+    return `${commands.TENANT_AUDITLOG_REPORT}`;
+  }
+
+  public get description(): string {
+    return 'Gets audit logs from the Office 365 Management API';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.startTime = typeof args.options.startTime !== 'undefined';
+    telemetryProps.endTime = typeof args.options.endTime !== 'undefined';
+    telemetryProps.contentType = args.options.contentType;
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['CreationTime', 'Operation', 'ClientIP', 'UserId', 'Workload'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Start retrieving Audit Log Report`);
+    }
+
+    this.tenantId = Utils.getTenantIdFromAccessToken(auth.service.accessTokens[auth.defaultResource].value);
+    this.getCompleteAuditReports(args, logger)
+      .then((res: AuditlogReport[]): void => {
+        logger.log(res);
+        if (this.verbose) {
+          logger.logToStderr(chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getCompleteAuditReports(args: CommandArgs, logger: Logger): Promise<AuditlogReport[]> {
+    return this.startContentSubscriptionIfNotActive(args, logger)
+      .then((): Promise<AuditContentList[]> => this.getAuditContentList(args, logger))
+      .then((auditContentLists: AuditContentList[]): void => this.getBatchedPromises(auditContentLists, 10))
+      .then((): Promise<void> => {
+        return new Promise<void>((resolve: () => void): void => { if (this.batchedPromises.length > 0) this.getBatchedAuditlogData(logger, 0, resolve); })
+      })
+      .then((): AuditlogReport[] => { return this.completeAuditReports.flat(2) });
+  }
+
+  private startContentSubscriptionIfNotActive(args: CommandArgs, logger: Logger): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Check whether subscription is active.`);
+    }
+
+    const subscriptionListEndpoint: string = 'activity/feed/subscriptions/list';
+    const requestOptions: any = {
+      url: `${this.serviceUrl}/${this.tenantId}/${subscriptionListEndpoint}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request.get<ActivityfeedSubscription[]>(requestOptions)
+      .then((subscriptionLists: ActivityfeedSubscription[]): boolean => {
+        return subscriptionLists.some(subscriptionList => subscriptionList
+          .contentType === (<any>AuditContentTypes)[args.options.contentType] && subscriptionList.status === 'enabled');
+      })
+      .then((hasActiveSubscription: boolean): Promise<void> => {
+        if (hasActiveSubscription) {
+          return Promise.resolve();
+        }
+        else {
```

Comment:
Since we're doing a `return` on line 124, we don't need the `else` here. That's the beauty of the early return: we avoid unnecessary nesting of code and improve its readability.

Reply:
Ahh Nice.... So silly from my end. I had not thought about that.. Thanks again...